### ISSUE TITLE
Harden session security through usage of CSRF 

### ISSUE THREAD

**Issue by autumnfound** [EclipseFdn/react-eclipsefdn-members#10]
While we already make use of SameSite user session cookies to harden our security, we should take additional steps to protect sensitive user data that can be stored in these forms. To do this, we should implement CSRF tokens with significant entropy into our requests to further protect our requests. As this request should be happening on the same machine, we should be able to use custom headers to secure these values in requests

The current implementation is not yet cryptographically hardened to an extent that meets [standard practices](https://portswigger.net/web-security/csrf/tokens). The aim for the implementation of a CSRF token should contain the following parts:

- [ ] cryptographic strength pseudo-random number generator (PRNG) seeded with the timestamp when it was created
- [ ] concatenate with a static secret
- [ ] generate individual tokens by concatenating its output with some user-specific entropy
- [ ] take a strong hash of the whole structure

**Comment by autumnfound** [comment]
This change is submitted for PR in the common package: https://gitlab.eclipse.org/eclipsefdn/webdev/eclipsefdn-api-common/-/merge_requests/3

**Comment by chrisguindon** [comment]
Upping the priority here. There is two part of this issue. 

We can only close this issue once this is implemented in the API and the react app.

More information about Avoiding CSRF Attacks with API Design:
http://www.thedreaming.org/2020/05/26/avoid-csrf-attacks-with-api-design/

**Comment by autumnfound** [comment]
In regards to SameSite:

https://stackoverflow.com/questions/42717210/samesite-cookie-in-java-application
TLDR: J EE doesn't implement SameSite, and its an NGINX proxy setting to set them to SameSite.

In regards to negative testing, I can make those changes as part of my testing changeset, as it will just be an additional test to ensure that CORS is being observed. That should take care of the 2 remaining items. The others are already present
